### PR TITLE
add frontier-pos-template

### PR DESCRIPTION
**Pull request by weimeme**
#### Project Abstract

This is a development template based on Polkadot's NPOS consensus, compatible with EVM. 
It aims to enable new developers or teams in the Polkadot ecosystem to quickly create EVM-compatible chains based on NPOS consensus.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [x] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (Polkadot AssetHub (DOT, USDC & USDT) address in the application and bank details via email, if applicable).
- [x] I understand that an agreed upon percentage of each milestone will be paid in vested DOT, to the Polkadot address listed in the application.
- [x] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by weimeme** [comment]
> **CLA Assistant Lite bot:** Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/T%26Cs.md) before we can accept your contribution. Please submit the following text as a separate comment:
> 
> I have read and hereby sign the Contributor License Agreement.
> 
> You can retrigger this bot by commenting **recheck** in this Pull Request

I have read and hereby sign the Contributor License Agreement.

File: applications/frontier-pos-template.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
+# frontier-pos-template
+
+- **Team Name:** ChainSupport
+- **Payment Details:**
+  - **Payment**: 0xcC1974d462B9F22F37D09c70F6023Ef9167EB3De (USDC)
```
Comment (semuelle) [inline applications/frontier-pos-template.md:5]: We need a Polkadot and AssetHub address. Payments on Ethereum are not offered anymore.

File: applications/frontier-pos-template.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+- https://github.com/JimYam
+- https://github.com/weimeme
+
+> Because our entire team of developers typically serves different public chain development teams, 
+> there is a requirement for a certain level of anonymity for our past GitHub accounts. If you need to know which specific projects we participated in, feel free to message me privately.
+## Development Status :open_book:
+
+### project
+https://github.com/ChainSupport/frontier-pos-template
```
Comment (semuelle) [inline applications/frontier-pos-template.md:127]: Can you elaborate what the status is?

File: applications/frontier-pos-template.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+### Team Code Repos
+
+- https://github.com/ChainSupport/frontier-pos-template
+
+- https://github.com/JimYam
+- https://github.com/weimeme
+
+> Because our entire team of developers typically serves different public chain development teams, 
+> there is a requirement for a certain level of anonymity for our past GitHub accounts. If you need to know which specific projects we participated in, feel free to message me privately.
```
Comment (semuelle) [inline applications/frontier-pos-template.md:123]: Can you please send us proof of your previous open source work to grants@web3.foundation?

File: applications/frontier-pos-template.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+### Overview
+
+- **Total Estimated Duration:** 1/3 months
+- **Full-Time Equivalent (FTE):**  6 FTE
+- **Total Costs:** 4,000USD
+> It will drive community discussions on further optimizing the public chain development process, allowing the community to collaboratively enhance the development experience at the application layer. 
+> The work involved could easily cost much more than this $4,000. We simply hope that it doesn't waste too much money before being widely adopted. In short, this initiative is worth pursuing.
+
+### Milestone 1 Example — Basic functionality
```
Comment (semuelle) [inline applications/frontier-pos-template.md:140]: This is template text, please adapt it to your project.

File: applications/frontier-pos-template.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. See the [delivery guidelines](https://grants.web3.foundation/docs/Support%20Docs/milestone-deliverables-guidelines#testing-guide) for details. |
+| **0d.** | Docker                    | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone.                                                                                                                                                                                                  |
+|     0e. | Article                   | We will publish an **article** that explains [...] (what was done/achieved as part of the grant). (Content, language, and medium should reflect your target audience described above.)                                                                                                                         |
```
Comment (semuelle) [inline applications/frontier-pos-template.md:154]: This is template text, please adapt it to your project.

File: applications/frontier-pos-template.md [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+|      3. | Substrate chain           | An EVM-compatible chain with NPOS consensus.                                                                                                                                                                                                                                                                                               |
+|     4.  | Friendly code structure                     | A developer-friendly code structure, similar to the one found in https://github.com/substrate-developer-hub/substrate-node-template.git.                                                                                                                                                                                                                        |
```
Comment (semuelle) [inline applications/frontier-pos-template.md:158]: This does not sound like a deliverable, but a feature of a deliverable. I suggest to remove it or make it part of one of the other deliverables.

File: applications/frontier-pos-template.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+- How did you identify these needs? Please provide evidence in the form of (scientific) articles, forum discussions, case studies, or raw data.
+  As long-term users of Substrate, we find it very painful to use the substrate-node-template. 
+Generally, the requirement is not to start with POA and then switch to NPOS, which causes many teams to spend a lot of effort on this upgrade.
+For new teams, they must understand the differences between the two and how to implement the technical upgrade, 
+leading to fear and lack of confidence among new developers, while older developers and teams waste time.
+
+- Are there any other projects similar to yours in the Substrate / Polkadot / Kusama ecosystem? If so, how is your project different? Please identify and assess any projects addressing the same need and explain how your project is distinct. Feel free to include applicable research data, statistics, or metrics.
+  https://github.com/substrate-developer-hub/substrate-node-template， No EVM compatibility and no NPOS.
+  https://github.com/substrate-developer-hub/frontier-node-template, No NPOS and the code structure is not developer-friendly.
```
Comment (semuelle) [inline applications/frontier-pos-template.md:77]: Can you explain how you want to improve the code structure to make it more developer-friendly?

**Review by semuelle — CHANGES_REQUESTED** [review]
Hi @weimeme, thanks for the application. Happy to support this, but there are some small issues I commented on below that need fixing/explaining.

**Comment by semuelle** [comment]
pinging @weimeme

**Comment by weimeme** [comment]
> pinging @weimeme

I apologize for the delayed response. We've been researching and coding recently. It turns out that creating such a template indeed requires a lot of effort and time from each development team, and it even necessitates highly experienced Substrate developers to accomplish it. I will review the application in the next few days and then update it.

**Comment by keeganquigley** [comment]
Hi @weimeme just a friendly ping for an update :)

**Review by Noc2 — CHANGES_REQUESTED** [review]
@weimeme  I will close the PR for now. Feel free to ping us, if I should reopen it.

**Comment by weimeme** [comment]
> @weimeme I will close the PR for now. Feel free to ping us, if I should reopen it.

@semuelle @keeganquigley @Noc2 @PieWol We apologize for the delay in responding. We have been busy developing the client for this project and communicating with more potential teams to understand their difficulties and needs. Currently, four teams are seeking our support to use Polkadot and Ethereum technologies to create their own solo chains. They are also following the progress of Polkadot 2.0, but their Substrate developers lack the experience to implement a solo chain that is compatible with both NPOS and EVM in a short time. We believe that supporting them while open-sourcing this technology and related discussions is very meaningful and significant for the large-scale adoption of Substrate technology.

Moreover, achieving full EVM compatibility requires more work. We found that in Blockscout, the transfer records of the balances module are still not displayed, which requires someone to try, identify, and solve the problems. We are very eager to receive initial support from W3F to explore this technology. At the same time, we are already promoting this technical solution to other teams, which will ensure continuous funding to maintain such a project in the future. I suggest we reopen this PR and discuss it further.

**Comment by weimeme** [comment]
The new grant application has been updated.

File: applications/frontier-pos-template.md (hunk 6 above)
Comment (weimeme) [reply applications/frontier-pos-template.md:158]: > This does not sound like a deliverable, but a feature of a deliverable. I suggest to remove it or make it part of one of the other deliverables.

The deliverable is a fully functional solo chain with NPOS and EVM compatibility.

**Comment by weimeme** [comment]
This is the recent development work we did for SuperEx: https://github.com/weimeme/scs-chain. The client that is compatible with both NPOS and EVM still requires us to do a lot of detailed development work, which has taken us a lot of time. We do not want other teams with similar needs to repeat this work in the future, as it is meaningless and unreasonable.

**Comment by Noc2** [comment]
Thanks for the updates here. However, I'm unable to reopen the PR since the master was force-pushed. Could you create a new PR?

**Comment by weimeme** [comment]
> Thanks for the updates here. However, I'm unable to reopen the PR since the master was force-pushed. Could you create a new PR?

@Noc2 Thank you very much for your response. I have submitted a new PR [https://github.com/w3f/Grants-Program/pull/2391](https://github.com/w3f/Grants-Program/pull/2391).

**Pull request closed without merging**